tests/ui/path_ends_with_ext.rs

```diff
@@ -0,0 +1,29 @@
+#![warn(clippy::path_ends_with_ext)]
+use std::path::Path;
+
+macro_rules! arg {
+    () => {
+        ".md"
+    };
+}
+
+fn test(path: &Path) {
+    path.ends_with(".md");
+    //~^ ERROR: this looks like a failed attempt at checking for the file extension
+
+    // some "extensions" are allowed by default
+    path.ends_with(".git");
+
+    // most legitimate "dotfiles" are longer than 3 chars, so we allow them as well
+    path.ends_with(".bashrc");
+
+    path.ends_with(arg!());
+}
```

Comment:
Could do with some cases like `..`, `./a`

Reply:
Good point, I guess we shouldn't lint on these since its not an extension. Maybe we could just check if all chars in the path (without the dot) match `char::is_alphanumeric` or smth 🤔